pkg/webhook/storageclass.go

```diff
@@ -27,152 +29,96 @@ import (
-		klog.Infof("Multishare Filestore CSI storage class detected")
-	}
-
-	// TBD: Fill the validation logic here.
-	return nil
-}
-
-func toV1AdmissionResponse(err error) *v1.AdmissionResponse {
+func rejectV1AdmissionResponse(err error) *v1.AdmissionResponse {
 	return &v1.AdmissionResponse{
 		Result: &metav1.Status{
 			Message: err.Error(),
 		},
 	}
 }
+
 func mutateStorageClass(ar v1.AdmissionReview) *v1.AdmissionResponse {
 	klog.Info("mutating storageClass")
 	reviewResponse := &v1.AdmissionResponse{
 		Allowed: true,
 		Result:  &metav1.Status{},
 	}
 
-	if !(ar.Request.Operation == v1.Update || ar.Request.Operation == v1.Create) {
+	if ar.Request.Operation != v1.Create {
 		return reviewResponse
 	}
 
 	raw := ar.Request.Object.Raw
-	oldRaw := ar.Request.OldObject.Raw
 
 	deserializer := codecs.UniversalDeserializer()
 	switch ar.Request.Resource {
 	case StorageClassV1GVR:
 		sc := &storagev1.StorageClass{}
 		if _, _, err := deserializer.Decode(raw, nil, sc); err != nil {
 			klog.Error(err)
-			return toV1AdmissionResponse(err)
-		}
-		oldSc := &storagev1.StorageClass{}
-		if _, _, err := deserializer.Decode(oldRaw, nil, oldSc); err != nil {
-			klog.Error(err)
-			return toV1AdmissionResponse(err)
+			return rejectV1AdmissionResponse(err)
 		}
 		klog.Infof("check patch for storageClass %s", sc.Name)
 		return applyV1StorageClassPatch(sc)
 	default:
-		err := fmt.Errorf("expect resource to be %s", StorageClassV1GVR)
+		err := fmt.Errorf("expect resource to be %v", StorageClassV1GVR)
 		klog.Error(err)
-		return toV1AdmissionResponse(err)
+		return rejectV1AdmissionResponse(err)
 	}
 }
 
 func applyV1StorageClassPatch(sc *storagev1.StorageClass) *v1.AdmissionResponse {
 	reviewResponse := &v1.AdmissionResponse{
 		Allowed: true,
+		Result:  &metav1.Status{},
 	}
 
 	if sc.Provisioner != FilestoreCSIDriver {
 		return reviewResponse
 	}
 
-	isMultishare, ok := sc.Parameters["multishare"]
-	if !ok || isMultishare == "false" || isMultishare == "False" {
+	isMultishare, ok := sc.Parameters[Multishare]
+	if !ok || strings.ToLower(isMultishare) == "false" {
 		return reviewResponse
 	}
 
-	if _, ok := sc.Parameters["instancePrefix"]; ok {
-		return reviewResponse
+	if strings.ToLower(isMultishare) != "true" {
+		return rejectV1AdmissionResponse(fmt.Errorf("the value for %q can only be True or False", Multishare))
+	}
+
+	tier, ok := sc.Parameters["tier"]
+	if !ok || tier != TierEnterprise {
+		return rejectV1AdmissionResponse(fmt.Errorf("mutlishare is only supported on %q tier instances", TierEnterprise))
 	}
 
-	scPatch := fmt.Sprintf(`[{"op":"add", "path":"/parameters/instancePrefix","value": "%s"}]`, sc.Name)
+	if instanceLabel, ok := sc.Parameters[InstanceStorageClassLabel]; ok {
+		if validateInstanceLabel(instanceLabel) {
+			return reviewResponse
+		} else {
+			return rejectV1AdmissionResponse(fmt.Errorf("%q must comply with https://cloud.google.com/filestore/docs/managing-labels#requirements", InstanceStorageClassLabel))
```

Comment:
lets not put the url. It would be good to frame the error msg similar to 
["label value %q is invalid (lowercase letter, digit, _ and - chars are allowed / 0-63 characters"](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/util/util.go#L106)

Reply:
done